README.md

```diff
@@ -1,2 +1,63 @@
 # radar-pipeline
-A python feature generation and visualzation package use with RADAR project data.
+
+A python feature generation and visualization package use with RADAR project data.
+
+## How to run
+
+**Note**: If you are using Windows, please install Spark and set environment variables as mentioned [here](https://sparkbyexamples.com/spark/apache-spark-installation-on-windows/) before going through the installation below. Specifically, you'll need to have the environment variables set, as given [here](https://spark.apache.org/docs/1.6.0/configuration.html#environment-variables).
+
+-   Clone the repository:
+
+    ```bash
+    $ git clone https://github.com/RADAR-base/radar-pipeline.git
+    ```
+
+-   Change the directory to `radar-pipeline`:
+
+    ```bash
+    $ cd radar-pipeline
+    ```
+
+-   Create a [virtual environment](https://virtualenv.pypa.io/en/latest/installation.html) and activate it.
+
+    -   Install the virtualenv package:
+
+        ```bash
+        $ python -m pip install --user virtualenv
+        ```
+
+    -   Create a python virtual environment:
+
+        ```bash
+        $ python -m virtualenv env
+        ```
+
+    -   Activate the virtual environment:
+
+        On Windows, run:
+
+        ```bash
+        $ .\env\Scripts\activate
+        ```
+
+        On Linux or MacOS, run:
+
+        ```bash
+        $ source ./env/bin/activate
+        ```
+
+-   Install the dependencies:
+
+    ```bash
+    $ pip install -r requirements.txt
+    ```
+
```

Comment:
How about specifying the downloading the submodule while cloning the repo itself through:
```bash
$ git clone --recurse-submodules https://github.com/RADAR-base/radar-pipeline.git
```

This is would need only one command to get all the files, and we can add a section in the readme about submodules in case somebody wants to refetch/update those.

The idea of automatically checking and downloading the submodules is good. What are your thoughts about downloading them **only** when the user runs a mock pipeline for the first time (like how HuggingFace libraries download the transformer models when the user runs the model configuration call the first time)?

Reply:
> How about specifying the downloading the submodule while cloning the repo itself through

Yes, that would work as well. 

> The idea of automatically checking and downloading the submodules is good. What are your thoughts about downloading them only when the user runs a mock pipeline for the first time 

Yes, that's the perfect solution. If the mock data is unavailable, it will download it otherwise would use the mock data locally.